### ISSUE TITLE
MouseEvent.x and .y have different types

### ISSUE THREAD

**Issue by tonyxty** [matplotlib/matplotlib#11496]
### Bug report

**Bug summary**

`MouseEvent.x` is an int but `MouseEvent.y` is a float.

**Code for reproduction**

```python
import numpy as np
import matplotlib.pyplot as plt


def onclick(event):
    print(event.x, event.y)
    print(type(event.x), type(event.y))


plt.imshow(np.zeros((32, 32)))
plt.gcf().canvas.mpl_connect('button_press_event', onclick)
plt.show()
```

**Actual outcome**

After randomly clicking on the image:
```
328 243.0
<class 'int'> <class 'numpy.float64'>
299 148.0
<class 'int'> <class 'numpy.float64'>
260 247.0
<class 'int'> <class 'numpy.float64'>
```

**Expected outcome**

```
328 243
<class 'int'> <class 'int'>
299 148
<class 'int'> <class 'int'>
260 247
<class 'int'> <class 'int'>
```

**Matplotlib version**

  * Operating system: Ubuntu 18.04
  * Matplotlib version: 2.2.2 (installed from pip)
  * Matplotlib backend (`print(matplotlib.get_backend())`): TkAgg
  * Python version: Python 3.6.5
  * Jupyter version (if applicable):
  * Other libraries:

**Comment by dstansby** [comment]
Is there any reason we can't cast to an int before y is extracted if it is always integer valued?

**Comment by QuLogic** [comment]
Probably not, but I suppose that theoretically you could get half-pixels on something like a HiDPI screen. But I don't know this for sure, and the docs never really made a commitment either way.

**Comment by anntzer** [comment]
At least the qt5 backend returns physical pixels here (we perform an explicit conversion in mouseEventCoords, so they should always be possible to return integers (which seems indeed better?).
On the other hand, the tk backend doesn't really support retina (https://github.com/matplotlib/matplotlib/issues/10388), so I'm not sure what's the exact situation there...

**Comment by anntzer** [comment]
(actually all backends currently share the inconsistency of returning x as int and y as float, so removed the tk label)

**Comment by tacaswell** [comment]
Lets cast y to be an int every where.

**Comment by TarasKuzyo** [comment]
I can submit a PR

**Comment by tacaswell** [comment]
Thanks @TarasKuzyo !  Please make sure you get all of the backends

**Comment by TarasKuzyo** [comment]
I was thinking of casting `x` and `y` to `int` in `LocationEvent`'s constructor, from which `MouseEvent` inherits. In such a way it will be backend-independent solution.

**Comment by tonyxty** [comment]
It would also be useful to specify the types of `x` and `y` in the documentation.

**Comment by QuLogic** [comment]
Fixed by #11530.
